### PR TITLE
Before after date filters

### PR DESCRIPTION
**Pull request by oblakeerickson**
Add `before_date` and `after_date` filters to filter topics by date in 'yyyy-mm-dd' format.

### CODE REVIEW

**Comment by oblakeerickson** [comment]
https://github.com/discourse/discourse-url-filters/pull/1 should be merged in first

File: spec/plugin_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  before { SiteSetting.url_filters_enabled = true }
+
+  describe 'Topic Queries' do
+    describe 'after_or_before_date' do
+      before do
+        @topic = Fabricate(:topic, created_at: 10.days.ago)
+        @topic2 = Fabricate(:topic, created_at: 8.days.ago)
+        @topic3 = Fabricate(:topic, created_at: 6.days.ago)
+        @user = Fabricate(:user)
+      end
```
Comment (CvX) [inline spec/plugin_spec.rb:15]:
```suggestion
      fab!(:topic) { Fabricate(:topic, created_at: 10.days.ago) }
      fab!(:topic2) { Fabricate(:topic, created_at: 8.days.ago) }
      fab!(:topic3) { Fabricate(:topic, created_at: 6.days.ago) }
      fab!(:user) { Fabricate(:user) }
```

(references to those objects will have to be be changed, e.g. `@user` -> `user`)

File: plugin.rb [hunk 2]
```diff
@@ -25,6 +26,28 @@
       results
     end
 
+    # After Date (yyyy-mm-dd)
+    TopicQuery.add_custom_filter(:after_date) do |results, topic_query|
+      if after_date = topic_query.options[:after_date]
+        if BASIC_DATE_REGEX.match(after_date)
+          parsed_after_date = Date.parse(after_date) rescue nil
+          results = results.where('topics.created_at > ?', parsed_after_date) if parsed_after_date
```
Comment (CvX) [inline plugin.rb:34]:
```suggestion
          begin
            parsed_after_date = Date.parse(after_date)
            results = results.where('topics.created_at > ?', parsed_after_date)
          rescue Date::Error
          end
```

File: plugin.rb [hunk 3]
```diff
@@ -25,6 +26,28 @@
...
+      end
+      results
+    end
+
+    # Before Date (yyyy-mm-dd)
+    TopicQuery.add_custom_filter(:before_date) do |results, topic_query|
+      if before_date = topic_query.options[:before_date]
+        if BASIC_DATE_REGEX.match(before_date)
+          parsed_before_date = Date.parse(before_date) rescue nil
+          results = results.where('topics.created_at < ?', parsed_before_date) if parsed_before_date
```
Comment (CvX) [inline plugin.rb:45]:
```suggestion
          begin
            parsed_before_date = Date.parse(before_date)
            results = results.where('topics.created_at < ?', parsed_before_date)
          rescue Date::Error
          end
```

**Review by CvX — CHANGES_REQUESTED** [review]

**Review by CvX — APPROVED** [review]

**Pull request merged**